### PR TITLE
Add Flagger

### PR DESCRIPTION
**Pull request by gpaul**
This PR also sets the istio prometheus service name, a change which is ported from work @GoelDeepak did on a branch of mine.

We'll squash the commits when we merge.

Chart:
https://github.com/mesosphere/charts/pull/188

https://jira.mesosphere.com/browse/DCOS-59422

### CODE REVIEW

File: templates/flagger.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.19.0"
+    docs.kubeaddons.mesosphere.io/flagger: "https://docs.flagger.app/"
+    values.chart.helm.kubeaddons.mesosphere.io/flagger: "https://raw.githubusercontent.com/weaveworks/flagger/0.19.0/charts/flagger/values.yaml"
+spec:
+  namespace: kubeaddons-flagger
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  cloudProvider:
+    - name: aws
+      enabled: true
```
Comment (jieyu) [inline templates/flagger.yaml]: Should we turn this off by default? @gpaul @GoelDeepak ?

**Comment by jieyu** [comment]
This PR also breaks konvoy e2e
https://teamcity.mesosphere.io/viewLog.html?buildId=2436038&buildTypeId=ClosedSource_MkeLite_MkeLiteE2e&tab=buildLog&_focus=10848

File: templates/flagger.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    - name: docker
+      enabled: true
+    - name: none
+      enabled: true
+  requires:
+    - matchLabels:
+        kubeaddons.mesosphere.io/name: istio
+  chartReference:
+    chart: flagger
+    repo: https://gpaul.github.io/charts/staging
```
Comment (hectorj2f) [inline templates/flagger.yaml]: @GoelDeepak  @shaneutt we should not deploy custom charts from our personal helm repositories

File: templates/flagger.yaml (hunk 1 above)
Comment (GoelDeepak) [reply templates/flagger.yaml]: I think that make sense as `istio` is disabled by default which is the requirement for `flagger`

File: templates/flagger.yaml (hunk 2 above)
Comment (hectorj2f) [reply templates/flagger.yaml]: that should use mesosphere charts repo

Comment (GoelDeepak) [reply templates/flagger.yaml]: I have this feeling that PR was merged before it was intented because the `flagger` charts are still in review

Comment (jieyu) [reply templates/flagger.yaml]: Let's revert then so that CI is unblocked.

Comment (hectorj2f) [reply templates/flagger.yaml]: 👍

Comment (shaneutt) [reply templates/flagger.yaml]: :+1:

**Pull request merged**